### PR TITLE
(NTRN-275) Feat: Testing denom checking in feerefunder

### PR DESCRIPTION
**Pull request by oldremez**
This PR adds really basic test for changes made in [this fix of Neutron code](https://github.com/neutron-org/neutron/pull/98)

The aim is to check if there is no ability to pay fees in denom that is not registered in MinFees parameter of the chain.

Ideally, this PR should be checked in a way where this test is run with main (`neutron_audit_oak_19_09_2022_fixes` atm) branch of Neutron first to ensure it fails and with `fix/fees-check-bypass` after that to ensure it passes.

### CODE REVIEW

File: src/testcases/simple.test.ts [hunk 1]
```diff
@@ -194,6 +194,56 @@ describe('Neutron / Simple', () => {
         ).rejects.toThrow(/invalid coins/);
       });
     });
+    describe('Fee in wrong denom', () => {
+      const uatom_ibc_denom =
+        'ibc/27394FB092D2ECCD56123C74F36E4C1F926001CEADA9CA97EA622B25F41E5EB2';
+      test('transfer some atoms to contract', async () => {
+        const uatom_amount = '1000';
```
[the remark this answers is not in the document]
Comment (oldremez) [reply src/testcases/simple.test.ts]: fixed 🙈

**Review by ratik — APPROVED** [review]

File: src/testcases/simple.test.ts [hunk 2]
```diff
@@ -195,6 +195,56 @@ describe('Neutron / Simple', () => {
...
+        expect(res.code).toEqual(0);
+
+        await waitBlocks(cm.sdk, 10);
+        const balances = await cm.queryBalances(contractAddress);
+        expect(
+          balances.balances.find((bal): boolean => bal.denom == uatomIBCDenom)
+            ?.amount,
+        ).toEqual(uatomAmount);
+      });
+      test('try to set fee in atoms', async () => {
```
Comment (albertandrejev) [inline src/testcases/simple.test.ts:219]: "try to set fee in IBC denominated atoms", but I am not sure that my variation is good enough.

File: src/testcases/simple.test.ts [hunk 3]
```diff
@@ -195,6 +195,56 @@ describe('Neutron / Simple', () => {
         ).rejects.toThrow(/invalid coins/);
       });
     });
+    describe('Fee in wrong denom', () => {
+      const uatomIBCDenom =
+        'ibc/27394FB092D2ECCD56123C74F36E4C1F926001CEADA9CA97EA622B25F41E5EB2';
```
Comment (foxpy) [inline src/testcases/simple.test.ts:200]: Would love to see a comment which describes what is encoded in this hashed denom.

File: src/testcases/simple.test.ts (hunk 2 above)
Comment (albertandrejev) [reply src/testcases/simple.test.ts:219]: agreed!

Comment (oldremez) [reply src/testcases/simple.test.ts]: Done

File: src/testcases/simple.test.ts (hunk 3 above)
Comment (oldremez) [reply src/testcases/simple.test.ts]: Comments suck so I added a calculation of the hash and checking if it's the right one

**Review by albertandrejev — APPROVED** [review]

**Review by foxpy — APPROVED** [review]

**Review by ratik — APPROVED** [review]

**Pull request merged**